pkg/controller/messaginginfra/broker/controller.go

```diff
@@ -108,10 +110,27 @@ func (b *BrokerController) ReconcileBrokers(ctx context.Context, logger logr.Log
 	}
 
 	// Update discoverable brokers
-	allHosts := make([]string, 0)
-	for host, _ := range hosts {
-		allHosts = append(allHosts, host)
+	brokerPods := corev1.PodList{}
+	err = b.client.List(ctx, &brokerPods, client.InNamespace(infra.Namespace), client.MatchingLabels(labels))
+	if err != nil {
+		return nil, err
 	}
+
+	allHosts := make([]state.Host, 0)
+	for expectedHost, _ := range hosts {
+		podIp := ""
+		for _, pod := range brokerPods.Items {
+			if pod.Status.Phase == corev1.PodRunning && pod.Status.PodIP != "" {
+				logger.Info("Found broker pod", "ip", pod.Status.PodIP)
+				if strings.HasPrefix(expectedHost, pod.Name) {
```

Comment:
It was simpler to check that the pod was a prefix of the full host (including statefulset name, namespace etc), rather than re-constructing the full host from the pod name only (remove -0, add that after pod name, add namespace etc.). Added a comment for it now.